wagtail/contrib/forms/forms.py

```diff
@@ -142,7 +141,7 @@ def formfields(self):
             # If the field hasn't been saved to the database yet (e.g. we are previewing
             # a FormPage with unsaved changes) it won't have a clean_name as this is
             # set in FormField.save.
-            clean_name = field.clean_name or get_field_clean_name(field.label)
+            clean_name = field.clean_name or field.get_field_clean_name()
```

Comment:
my one question on the PR: can we safeguard against bad values coming from the overridden `field.get_field_clean_name()`?

Reply:
What would you define as a bad value?

As in a non-string?

Reply:
OK after some digging, I think that we can allow pretty much any string, the ascii safe is probably a hangover from supporting ids in IE11.

We need to meet two criteria
* Valid JSON Key (which can be any UTF string) https://tools.ietf.org/id/draft-ietf-json-rfc4627bis-09.html#rfc.section.8.2
* Valid HTML attribute in HTML 5 - this just must be any non-whitespace character, there would be some nuance if you wanted to also write CSS for selecting something like an emoji ID but the page will render and the form should submit fine. See https://mathiasbynens.be/notes/html5-id-class & https://html.spec.whatwg.org/multipage/dom.html#the-id-attribute & https://html.spec.whatwg.org/multipage/dom.html#attributes (specifically for the name attribute)

So with that being said, we can maybe update the function docstring to something like.

```python
def get_field_clean_name(label):
    """
    Converts a user entered field label to a string that is safe to use for a HTML attribute and JSON key.
    The result will be used as the internal key (clean name) for the field and the form field name.
    """
    return safe_snake_case(label)
```

We can still add a check that the resolved value is not empty though, but I think trying to validate the tiny edge cases of risky encoding will create more problems than it solves.

Reply:
sounds like a plan! great 🕵🏼 work

Reply:
Done.